app/src/main/java/com/github/se/travelpouch/ui/navigation/SwipePager.kt

```diff
@@ -0,0 +1,127 @@
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.outlined.ArrowBack
+import androidx.compose.material.icons.filled.AccountCircle
+import androidx.compose.material.icons.filled.DateRange
+import androidx.compose.material.icons.filled.Folder
+import androidx.compose.material.icons.filled.Settings
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.NavigationBar
+import androidx.compose.material3.NavigationBarItem
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import com.github.se.travelpouch.model.activity.ActivityViewModel
+import com.github.se.travelpouch.model.dashboard.CalendarViewModel
+import com.github.se.travelpouch.ui.dashboard.CalendarScreen
+import com.github.se.travelpouch.ui.dashboard.TravelActivitiesScreen
+
+// credit to : https://www.youtube.com/watch?v=inrGyNJbZaI&t=620s
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun SwipePager(
+    navigationActions: NavigationActions,
+    activityViewModel: ActivityViewModel,
+    calendarViewModel: CalendarViewModel
+) {
+  val listOfTopLevelDestinationForSwipe =
+      listOf(TopLevelDestinations.ACTIVITIES, TopLevelDestinations.CALENDAR)
+
+  val listOfTopLevelDestination =
+      listOf(
+          TopLevelDestinations.ACTIVITIES, TopLevelDestinations.CALENDAR, TopLevelDestinations.MAP)
+
+  val pagerState = rememberPagerState(initialPage = 0) { listOfTopLevelDestinationForSwipe.size }
+
+  var selectedScreen by remember { mutableIntStateOf(pagerState.currentPage) }
+
+  LaunchedEffect(selectedScreen) { pagerState.scrollToPage(selectedScreen) }
+
+  LaunchedEffect(pagerState.currentPage) { selectedScreen = pagerState.currentPage }
+
+  Scaffold(
+      modifier = Modifier.testTag("pagerSwipe"),
+      topBar = {
+        TopAppBar(
+            modifier = Modifier.testTag("topBar"),
+            title = {
+              Text(
+                  listOfTopLevelDestinationForSwipe[selectedScreen].textId,
+                  Modifier.testTag("travelTitle"))
+            },
+            navigationIcon = {
+              IconButton(
+                  onClick = { navigationActions.navigateTo(Screen.TRAVEL_LIST) },
+                  modifier = Modifier.testTag("goBackButton")) {
+                    Icon(
+                        imageVector = Icons.AutoMirrored.Outlined.ArrowBack,
+                        contentDescription = "Back")
+                  }
+            },
```

Comment:
While it is good that the topbar and bottom bar stay static and simply change names, when pressing the going back button on calendar one goes back to travel list screen. This is not the same behaviour as before.

Reply:
This is a good idea. I did it